NEWS.md

```diff
@@ -1,3 +1,79 @@
+# 1.x
+
+TODO:
+- Word about new organization and repo move?
+- Word about this being release of master as of now etc?
+
+Changes
+
+- Use decimal number literals to preserve precision. Comparsion operations respects precision but arithmetic operations might truncate or .  #1752 @leonid-s-usov
+  ```sh
+  # precision is preserved
+  $ jq -n '100000000000000000'
+  100000000000000000
+  # comparsion respects precision (this is false in JavaScript)
+  $ jq -n '100000000000000000 < 100000000000000001'
+  true
+  # arithmetic operations might truncate (same as JavaScript)
+  $ jq -n '100000000000000000+10'
+  100000000000000020
+  ```
+- Allow `if` without `else`-branch. When skipped thee `else`-branch will be `.` (identity). 4f6045a94e0519ec58210bbd58ece442c32468e1 @chancez
+   ```sh
+   # convert 1 to "one" otherwise keep as is
+   $ jq -n '1,2 | if . == 1 then "one" end'
+   "one"
+   2
+   # behaves the same as
+   $ jq -n '1,2 | if . == 1 then "one" else . end'
+   "one"
+   2
+   ```
+   Note that optional `else` currently does not work with `elif`, hopefully will be addressed in later release (issue [#2481: Elif and optional else not working](https://github.com/jqlang/jq/issues/2481))
+- Allow use of `$binding` as key in object literals. 8ea4a55 @nicowilliams
+  ```sh
+  $ jq -n '"a" as $key | {$key: 123}'
+  {
+    "a": 123
+  }
+  # previous parentheses was needed
+  $ jq -n '"a" as $key | {($key): 123}'
+  {
+    "a": 123
+  }
+  ```
+- Last output value can now control exit code using `--exit-code`/`-e`. #1697 @ryo1kato
+  ```sh
+  # true-ish last output value exits with zero
+  $ jq -ne true ; echo $?
+  true
+  0
+  # false-ish last output value (false and null) exits with 1
+  $ jq -ne false ; echo $?
+  false
+  1
+  # no output value exists with 4
+  $ jq -ne empty ; echo $?
+  4
+  ```
+- Allow dot between chained indexes when using `.["index"]` #1168 @nicowilliams
+  ```sh
+  $ jq -n '{"a": {"b": 123}} | .a["b"]'
+  123
+  # now this works also
+  $ jq -n '{"a": {"b": 123}} | .a.["b"]'
+  123
+  ```
+- Speed up and refactor some builtins, also remove `scalars_or_empty/0`. #1845 @muhmuhten
+- Add `--binary`/`-b` on Windows for binary output. To get `\n` instead of `\r\n` line endings. 0dab2b1 @nicowilliams
+- Add `--nul-output`/`-0` for null (zero byte) separated output. #1990 @pabs3
+  ```sh
+  $ jq -n0 '1,2,3' | xxd
+  00000000: 3100 3200 3300                           1.2.3.
+  ```
```

Comment:
This probably shouldn't be mentioned? #2535

Reply:
This one is a bit confusing, #1271 was not been merged so jq master seems to still have it. Also the shorthand `-0` does not work as jq's argument parser reads it as a negative number i think.

Reply:
If `-0` doesn't work, then it shouldn't be documented. And if you're likely to remove the other flag, then even if it technically exists, I'd suggest not announcing it, at least, without some big hazard warnings that it's likely to be removed and pointing to an alternative.

Reply:
Now i remember. Short flag `-0` do work but only if used in combination with other short flags like in `-n0`. This is very confusing, so maybe we should only use long flag for this or come up with a different short flag? or fix the argument parsing code somehow?

I do think null separated output can be useful (but can be dangerous with `-r`) so personally i would like to keep it in some form.

@itchyny what do you think? and i also noticed gojq has no issue with this:
```sh
$ gojq -n -0 123
123% # (% is zsh showing there is no new line)
$ jq -n -0 123
-0
```

Reply:
Wow, that's indeed unfortunate.

In the interim, I'd definitely suggest not documenting it, and maybe even removing it until the behavior can be implemented in a less confusing manner.

I personally am a fairly heavy `-z` / `-0` user of various things (e.g. `git ls-files -z|xargs -0` or `find . -print0|xargs -0` or things where my own code consumes/produces `nul` delimited content).

Reply:
Seem like @itchyny found and old PR to fix the argument parsing https://github.com/jqlang/jq/pull/2235

Reply:
Yeah, I merged the PR so now we can use `-0` with the master version.
```bash
 $ jq --version
jq-1.6-180-g6c5c8f2

 $ jq -n -0 1,2,3 | xargs -0 -n1
1
2
3
```

Reply:
Do you want to change the PR reference to also include that PR?

Reply:
Yes, that's better to be included in this list.

Reply:
Yes will fix and also steal @itchyny example once im at home

Reply:
Using short flag combination is user's preference, so I think it's better not to use it in the release note. Maybe more example which cannot be archived before would help user know how it is useful?
```sh
$ jq -n -0 '1,2,3' | xxd
00000000: 3100 3200 3300                           1.2.3.
# can be used with xargs -0
$ jq -n -0 '"a b c", "d\ne\nf"' | xargs -0 printf '%q\n'
'a b c'
'd'$'\n''e'$'\n''f'
# can be used with read -d ''
$ while IFS= read -r -d '' json; do
>   jq '.name' <<< "$json"
> done < <(jq -n -0 '{name:"a b c"},{name:"d\ne\nf"}')
"a b c"
"d\ne\nf"
```

Reply:
Good idea, added your examples, changed my 1,2,3 examples to us a,b,c instead, seems better. Also changed to correct say that zero byte is after each output not between.

Could you clarify what you mean by "Using short flag combination is user's preference, so I think it's better not to use it in the release note"? use `-n -0`  not `-n0` in  release notes?